toolflow/boot/generate_boot_image.sh

```diff
@@ -237,37 +245,78 @@ build_linux() {
 	if [[ $ARCH != arm64 ]]; then
 		if [[ ! -e $DIR/linux-xlnx/arch/arm/boot/Image ]]; then
 			echo "Building linux $VERSION .."
+            #create tapasco specific defconfig
 			DEFCONFIG=tapasco_zynq_defconfig
-			CONFIGFILE="$SCRIPTDIR/configs/tapasco_zynq_defconfig"
-			cp $CONFIGFILE $DIR/linux-xlnx/arch/arm/configs/ ||
-				return $(error_ret "$LINENO: could not copy config")
+            cp $DIR/linux-xlnx/arch/arm/configs/xilinx_zynq_defconfig $DIR/linux-xlnx/arch/arm/configs/$DEFCONFIG ||
+				return $(error_ret "$LINENO: could not duplicate zynq defconfig")
+
+            CONFIGFILE="$DIR/linux-xlnx/arch/arm/configs/$DEFCONFIG"
+            echo "Adding tapasco specific config options to config file"
+            append_if_not_exists 'CONFIG_LOCALVERSION="-tapasco"' $CONFIGFILE
+			append_if_not_exists "CONFIG_DEFAULT_HOSTNAME=\"$BOARD\"" $CONFIGFILE
+
 			cd $DIR/linux-xlnx
 			make CROSS_COMPILE=$CROSS_COMPILE ARCH=arm $DEFCONFIG ||
 				return $(error_ret "$LINENO: could not make defconfig")
 			make CROSS_COMPILE=$CROSS_COMPILE ARCH=arm -j $JOBCOUNT ||
 				return $(error_ret "$LINENO: could not build kernel")
+			#works better when doing on host and copying later
+			make CROSS_COMPILE=$CROSS_COMPILE ARCH=arm modules_install INSTALL_MOD_PATH=.
 		else
 			echo "$DIR/linux-xlnx/arch/arm/boot/Image already exists, skipping."
 		fi
 	else
 		if [[ ! -e $DIR/linux-xlnx/arch/arm64/boot/Image ]]; then
 			echo "Building linux $VERSION for arm64.."
+            #name for tapasco specific defconfig file
 			DEFCONFIG=tapasco_zynqmp_defconfig
-			CONFIGFILE="$SCRIPTDIR/configs/tapasco_zynqmp_defconfig"
-			cp $CONFIGFILE $DIR/linux-xlnx/arch/arm64/configs/ ||
-				return $(error_ret "$LINENO: could not copy config")
+            #base tapasco specific config on current version of zynqmp defconfig
+			cp $DIR/linux-xlnx/arch/arm64/configs/xilinx_zynqmp_defconfig $DIR/linux-xlnx/arch/arm64/configs/$DEFCONFIG ||
+				return $(error_ret "$LINENO: could not duplicate zynqmp defconfig")
+
+			CONFIGFILE="$DIR/linux-xlnx/arch/arm64/configs/$DEFCONFIG"
+			echo "Adding tapasco specific config options to config file"
+			#add tapasco specific config options to new defconfig file
+			append_if_not_exists 'CONFIG_LOCALVERSION="-tapasco"' $CONFIGFILE
+			append_if_not_exists "CONFIG_DEFAULT_HOSTNAME=\"$BOARD\"" $CONFIGFILE
+			append_if_not_exists 'CONFIG_VFIO=y' $CONFIGFILE
+			append_if_not_exists 'CONFIG_VFIO_PLATFORM=y' $CONFIGFILE
+			append_if_not_exists 'CONFIG_VFIO_IOMMU_TYPE1=y' $CONFIGFILE
+			append_if_not_exists 'CONFIG_ARM_SMMU=y' $CONFIGFILE
+			append_if_not_exists 'CONFIG_USB_RTL8152=y' $CONFIGFILE
+			append_if_not_exists 'CONFIG_USB_USBNET=y' $CONFIGFILE
```

Comment:
Why don't you put this into the `configs/tapasco_zynqmp_defconfig` config file?

Reply:
the current `configs/tapasco_zynqmp_defconfig` file does not work with newer kernel versions. To always have the correct config as a basis, I copy `xilinx_zynqmp_defconfig` from the fetched linux source and only append tapasco specific options. To still support older version there should probably be a version check here though

Reply:
We should not overcomplicate things, just leave it as it is.